src/Akka.Persistence.Sql/Db/AkkaPersistenceDataConnectionFactory.cs

```diff
@@ -42,17 +42,19 @@ public AkkaPersistenceDataConnectionFactory(IProviderConfig<JournalTableConfig>
             fmb.Build();
 
             _useCloneDataConnection = config.UseCloneConnection;
-            _opts = new DataOptions()
+            var opts =new DataOptions()
                 .UseConnectionString(config.ProviderName, config.ConnectionString)
                 .UseMappingSchema(mappingSchema);
+            _opts = opts;
 
             if (config.ProviderName.ToLower().StartsWith("sqlserver"))
                 _policy = new SqlServerRetryPolicy();
-
+            
+            
             _cloneConnection = new Lazy<AkkaDataConnection>(
                 () => new AkkaDataConnection(
-                    _opts.ConnectionOptions.ProviderName,
-                    new DataConnection(_opts)));
+                    opts.ConnectionOptions.ProviderName,
+                    new DataConnection(opts)));
```

Comment:
subtle thing; now we are capturing just the `DataOptions` reference rather than the factory itself...

IDK how much the GC *really* cares but it sounds like the sort of thing that can annoy a cycle detector,

Reply:
LGTM